### PR TITLE
Add source to span details

### LINKED ISSUE #3711: Dashboard doesn't show the name of the component firing telemetry
image

In this case the telemetry is from httpclient - I know that because of the URL and the Kind is client, but we are not showing the provider for the telemetry.
 
Its included in the output as the ActivitySourceName:
 
230Activity.TraceId:            6d064cbf35d48958be1237d7aa9e1bd5
231Activity.SpanId:             4b046ffb0fb88c70
232Activity.TraceFlags:         Recorded
233Activity.ParentSpanId:       785f37b34b83d616
234Activity.ActivitySourceName: System.Net.Http
235Activity.DisplayName:        GET
236Activity.Kind:               Client
237Activity.StartTime:          2024-04-15T16:51:20.2069090Z
238Activity.Duration:           00:00:00.2837946
239Activity.Tags:
240    http.request.method: GET
241    server.address: localhost
242    server.port: 7421
243    url.full: https://localhost:7421/weatherforecast
244    network.protocol.version: 1.1
245    http.response.status_code: 200
246Resource associated with Activity:
247    service.name: webfrontend
248    service.instance.id: webfrontend-0emuqt8
249    telemetry.sdk.name: opentelemetry
250    telemetry.sdk.language: dotnet
251    telemetry.sdk.version: 1.8.0

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/dotnet/aspire/issues/3711

While making other repository changes I noticed that the source was on the trace instead of span. Moved it, and added the UI to display a context section to span details. The new section includes the span's source (in .NET land, this is the activity source name such as `Microsoft.AspNetCore` or `System.Net.Http`), parent id and trace id.

image

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/4703)

### CODE REVIEW

**Comment by JamesNK** [comment]
Please review 🙏

File: tests/Aspire.Dashboard.Tests/TelemetryRepositoryTests/TraceTests.cs [hunk 1]
```diff
@@ -196,17 +271,15 @@ public void AddTraces_Traces_MultipleOutOrOrder()
             {
                 AssertId("1", trace.TraceId);
                 AssertId("1-1", trace.FirstSpan.SpanId);
-                AssertId("", trace.FirstSpan.ScopeName);
+                Assert.Equal("", trace.FirstSpan.Scope.ScopeName);
```
Comment (radical) [inline tests/Aspire.Dashboard.Tests/TelemetryRepositoryTests/TraceTests.cs:274]: Just curious - why the change from `AssertId` to `Assert.Equal`?

**Review by radical — APPROVED** [review]
I'm not really familiar with the code. But diff seems to do what the PR description says, and makes sense to me - whatever I could understand!

File: tests/Aspire.Dashboard.Tests/TelemetryRepositoryTests/TraceTests.cs (hunk 1 above)
Comment (JamesNK) [reply tests/Aspire.Dashboard.Tests/TelemetryRepositoryTests/TraceTests.cs:274]: `AssertId` does some work to convert hex encoded bytes to a UTF8 string. It was wrongly being used here. It only worked before the scope name happened to be an empty string.

File: src/Aspire.Dashboard/Components/Controls/SpanDetails.razor.cs [hunk 2]
```diff
@@ -33,4 +38,24 @@ private bool ApplyFilter(SpanPropertyViewModel vm)
         return vm.Name.Contains(_filter, StringComparison.CurrentCultureIgnoreCase) ||
             vm.Value?.Contains(_filter, StringComparison.CurrentCultureIgnoreCase) == true;
     }
+
+    protected override void OnParametersSet()
+    {
+        _contextAttributes =
+        [
+            new KeyValuePair<string, string>("Source", ViewModel.Span.Scope.ScopeName)
```
Comment (adamint) [inline src/Aspire.Dashboard/Components/Controls/SpanDetails.razor.cs:46]: is there a reason these are hardcoded?

Comment (JamesNK) [reply src/Aspire.Dashboard/Components/Controls/SpanDetails.razor.cs:46]: Are you asking why the names aren't in resource files for translation? None of the "known" property names in the details screens are translated because none of the properties that come from OTEL are translated.

Another example: https://github.com/dotnet/aspire/blob/7d5a03251a9e7e3099e87ee34c9499dde3095288/src/Aspire.Dashboard/Components/Controls/StructuredLogDetails.razor.cs#L53-L64

**Comment by radical** [comment]
`main` rolling build broke:
```
D:\a\_work\1\s\tests\Aspire.Dashboard.Tests\TelemetryRepositoryTests\TraceTests.cs(147,13): error CS0117: 'GetTracesRequest' does not contain a definition for 'ApplicationServiceId' [D:\a\_work\1\s\tests\Aspire.Dashboard.Tests\Aspire.Dashboard.Tests.csproj]
##[error]tests\Aspire.Dashboard.Tests\TelemetryRepositoryTests\TraceTests.cs(147,13): error CS0117: (NETCORE_ENGINEERING_TELEMETRY=Build) 'GetTracesRequest' does not contain a definition for 'ApplicationServiceId'
D:\a\_work\1\s\tests\Aspire.Dashboard.Tests\TelemetryRepositoryTests\TraceTests.cs(145,47): error CS9035: Required member 'GetTracesRequest.ApplicationKey' must be set in the object initializer or attribute constructor. [D:\a\_work\1\s\tests\Aspire.Dashboard.Tests\Aspire.Dashboard.Tests.csproj]
##[error]tests\Aspire.Dashboard.Tests\TelemetryRepositoryTests\TraceTests.cs(145,47): error CS9035: (NETCORE_ENGINEERING_TELEMETRY=Build) Required member 'GetTracesRequest.ApplicationKey' must be set in the object initializer or attribute constructor.
```

**Comment by JamesNK** [comment]
I included a fix in another PR: [`5acbf8f` (#4764)](https://github.com/dotnet/aspire/pull/4764/commits/5acbf8fd56629b6d6e70225429a9019518c83b2c)

**Comment by radical** [comment]
> I included a fix in another PR: [`5acbf8f` (#4764)](https://github.com/dotnet/aspire/pull/4764/commits/5acbf8fd56629b6d6e70225429a9019518c83b2c)

Cool, assuming that it is ready to merge once CI is green. Thanks!
